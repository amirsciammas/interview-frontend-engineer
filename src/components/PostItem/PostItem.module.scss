.postItem {
  padding: 30px;
  border-radius: 6px;
  background-color: white;
  box-shadow: 0px 3px 5px -1px rgba(0, 0, 0, 0.2),
    0px 6px 10px rgba(0, 0, 0, 0.14), 0px 1px 18px rgba(0, 0, 0, 0.12);
}

.postItem:hover {
  outline: 2px solid grey;
}

.title {
	font-weight: bold;
	color: indigo;
}
