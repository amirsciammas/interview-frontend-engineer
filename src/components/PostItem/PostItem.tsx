import React, { FC } from "react";
import { Post } from "../../interfaces/PostType";
import styles from "./PostItem.module.scss";

interface PostItemProps {
  post: Post;
}

const PostItem: FC<PostItemProps> = ({ post }) => (
  <div className={styles.postItem} data-testid="PostItem">
    <h4 className={styles.title} data-testid="PostItem.title">
      {post.title}
    </h4>
    <div data-testid="PostItem.body">{post.body}</div>
  </div>
);

export default PostItem;
