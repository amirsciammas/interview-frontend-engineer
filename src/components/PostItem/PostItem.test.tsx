import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import PostItem from "./PostItem";
import renderer from "react-test-renderer";

const post = {
  userId: 0,
  id: 0,
  title: "",
  body: "",
};

describe("<PostItem />", () => {
  test("it should mount", () => {
    render(<PostItem post={post} />);

    const postItem = screen.getByTestId("PostItem");
    expect(postItem).toBeInTheDocument();
  });

  test("renders user data", () => {
    render(<PostItem post={post} />);
    const postItemTitle = screen.getByTestId("PostItem.title");
    expect(postItemTitle.textContent).toBe(post.title);

    const postItemBody = screen.getByTestId("PostItem.body");
    expect(postItemBody.textContent).toBe(post.body);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<PostItem post={post} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
