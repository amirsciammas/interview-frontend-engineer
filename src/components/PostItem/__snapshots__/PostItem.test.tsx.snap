// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItem /> renders correctly 1`] = `
<div
  className="postItem"
  data-testid="PostItem"
>
  <h4
    className="title"
    data-testid="PostItem.title"
  >
    
  </h4>
  <div
    data-testid="PostItem.body"
  >
    
  </div>
</div>
`;
